src/policy/compiler.rs

```diff
@@ -1377,6 +1378,46 @@ mod tests {
             ]
         );
     }
+
+    #[test]
+    fn compile_thresh() {
+        let (keys, _) = pubkeys_and_a_sig(21);
+
+        // Up until 20 keys, thresh should be compiled to a multi no matter the value of k
+        for k in 1..4 {
+            let small_thresh: BPolicy = policy_str!(
+                "thresh({},pk({}),pk({}),pk({}))",
+                k,
+                keys[0],
+                keys[1],
+                keys[2]
+            );
+            let small_thresh_ms: SegwitMiniScript = small_thresh.compile().unwrap();
+            let small_thresh_ms_expected =
+                ms_str!("multi({},{},{},{})", k, keys[0], keys[1], keys[2]);
+            assert_eq!(small_thresh_ms, small_thresh_ms_expected);
+        }
+
+        // Above 20 keys, thresh is compiled to a combination of and()s if it's a N of N,
+        // and to a ms thresh otherwise.
+        // FIXME: k = 1 (or 2, 3) doesn't work !
```

Comment:
What does "doesn't work" mean?

Reply:
I know you saw this issue. But for completeness of future viewers( refer #114 )

Reply:
Will link the issue instead, and maybe remove the "FIXME" as it seems we're not going to fix it ?

Reply:
Yes, that would be great.